src/test/ui/mir/simplify-branch-same.rs

```diff
@@ -0,0 +1,21 @@
+// Regression test for SimplifyBranchSame miscompilation.
+// run-pass
+
+macro_rules! m {
```

Comment:
is there a reason for this macro here?

Reply:
The optimization uses a form of statement equality involving either source locations or spans.

Reply:
That sounds pretty concerning, given that proc macros are able to synthesize equivalent spans whenever they want on any expression -- I have not looked at code though, so maybe that's mitigated.

Reply:
In other words it is much more conservative than necessary. Which is not great either, since it makes it very hard to test.

Reply:
> That sounds pretty concerning,

It does, especially since i can't tell how/why the code would do this anywhere.

Reply:
ah, that's why I wasn't able to trigger this opt in my own test case :sweat_smile: that sounds really suboptimal